py-polars/polars/_utils/construction/dataframe.py

```diff
@@ -1040,6 +1040,14 @@ def pandas_to_pydf(
     include_index: bool = False,
 ) -> PyDataFrame:
     """Construct a PyDataFrame from a pandas DataFrame."""
+    stringified_cols = [str(col) for col in data.columns]
```

Comment:
This first converts to a list and then to a set. Can we do that in a single pass? E.g. construct a set directly and then check the set length with the dataframe width.